x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPlugin.java

```diff
@@ -172,11 +172,11 @@ private void registerGeoShapeGridAggregators(ValuesSourceRegistry.Builder builde
                 if (getLicenseState().checkFeature(XPackLicenseState.Feature.SPATIAL_GEO_GRID)) {
                     final GeoGridTiler tiler;
                     if (geoBoundingBox.isUnbounded()) {
-                        tiler = new GeoTileGridTiler();
+                        tiler = new UnBoundedGeoTileGridTiler(precision);
```

Comment:
I have struggling a bit over this one. There are two ways to approach this - make tiler a singleton and pass everything as parameters to calls or pass or agg-level parameter here. I am fine with ether, the mix was a bit weird, so I think it is a good change.